docs/src/datamanagment/metadata-search.md

```diff
@@ -0,0 +1,360 @@
+is available, or none of it is. Commits are processed sequentially: a child commit will only be processed after its parent
+has been fully ingested.
+
+!!! tip 
+    To check whether the most recent state of your branch is available for metadata search queries, check if the following 
+    query returns results:
+    ```sql
+    USE "<repo>-metadata.<branch>.system";
+    SELECT * FROM object_metadata
+    WHERE commit_id = <head_commit> -- Replace with the head commit ID of the branch you are looking at 
+    LIMIT 1;
+    ```
+
+## Getting Started
+
+### Configuration
+
+lakeFS Metadata Search runs as a separate service that integrates with your lakeFS server.  
+
+If you are **self-hosting lakeFS Enterprise**:
+
+1. [Contact us](https://lakefs.io/lp/metadata-search/) to enable the feature.  
+2. Add the configuration below to your Helm values file.
+3. Install or upgrade the Helm chart with the updated configuration.
+
+If you are using **lakeFS cloud**:
+
+[Contact us](https://lakefs.io/lp/metadata-search/) to enable the feature. We’ll request the information included in the 
+sample configuration below.
+
+The Metadata Search service requires:
+
+* **lakeFS server connection settings**: so the service can communicate with your lakeFS instance.
+* **Metadata-specific settings**: to control how metadata is captured and which repositories and branches are searchable.
+
+#### Configuration Reference
+
+* `lakefs.endpoint` `(string : "")`- the lakeFS server endpoint
+* `lakefs.access_key_id` `(string : "")`- a lakeFS access key 
+* `lakefs.secret_access_key` `(string : "")`- a lakeFS secret access key
+* `metadata_settings.since_epoch` `(string : "")`- ISO 8601 timestamp (e.g., `2ș24-01-01T00:00:00Z`) that sets the earliest
+point in time from which to process commits for metadata extraction. If omitted, metadata will be captured from the time 
+the branch was created.
+* `metadata_settings.max_commits` `(int : 0)` - The maximum number of commits to process per searchable branch.
+  Uses 0 by default that disables the limit.
+* `metadata_settings.repositories` `(map[string]string[] : {)` - A mapping of repositories and the branches in each where metadata 
+search should be enabled. You can specify full branch names or use branch prefixes for flexibility.
+
+!!! note
+    Metadata search is disabled by default. You must explicitly configure which repositories and branches to include.
+
+!!! tip
+    Use branch name prefixes (e.g., `feature-*`) to reduce the need for manual updates when new branches are added.
+
+#### Sample Configuration
+
+!!! example
+    ```yaml
+    lakefs:
+      endpoint: "https://example.lakefs.io"
+      access_key_id: "AKIAIOSFOLEXAMPLE"
+      secret_access_key: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+    metadata_settings:
+      since_epoch: "2024-01-01T00:00:00Z"
+      max_commits: 100
+      repositories:
+        "example-repo-1":
+          - "main"
```

Comment:
This is not docs related - it's a design comment which I feel strongly about:
The heirarchy here is too implicit and will not support tags and arbitrary commits.
I suggest:

```yaml
...
repositories:
  example-repo-1:
    branches:
      - main
      - dev
```

This would later allow adding `tags` or `commits` while being backwards compatible.